assembl/static2/js/app/components/debate/tableOfIdeas/ideas.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { Translate } from 'react-redux-i18n';
+import classnames from 'classnames';
+import { Grid } from 'react-bootstrap';
+import IdeasLevel from './ideasLevel';
+import IdeasLevelMobile from './ideasLevelMobile';
+import { SMALL_SCREEN_WIDTH } from '../../../constants';
+
+const PageTitle = () => (
+  <div className="title-section">
+    <div className="title-hyphen">&nbsp;</div>
+    <h1 className="dark-title-1">
+      <Translate value="debate.survey.themesTitle" />
+    </h1>
+  </div>
+);
+
+class Ideas extends React.Component {
+  constructor(props) {
+    super(props);
+    const { rootIdeaId } = this.props;
+    this.state = { selectedIdeasId: [rootIdeaId], selectedIdeaIndex: 0, ideaLevel: 0, goBack: false };
+    this.getIdeaChildren = this.getIdeaChildren.bind(this);
+    this.setSelectedIdeas = this.setSelectedIdeas.bind(this);
+    this.getIdeaParents = this.getIdeaParents.bind(this);
+    this.goBackToParents = this.goBackToParents.bind(this);
```

Comment:
You can use arrow functions to avoid having to bind to "this"

Reply:
Flow n'est pas utilisé dans ce composant. C'est pourquoi j'ai gardé le même pathern. C'est un vieux truc. Je vais donc mettre les annotations flow :(